SerialPrograms/Source/PokemonHome/Programs/PokemonHome_NationalDexSorter.cpp

```diff
@@ -0,0 +1,1489 @@
+            if(itemNo-i>=7-itemOffset&&itemNo>=7)
+            {
+                pbf_press_dpad(context, DPAD_RIGHT, 5, leSpeed);
+                i+=7;
+            }
+            else if(itemNo-i==6-itemOffset&&itemNo>=6)
+            {
+                pbf_press_dpad(context, DPAD_RIGHT, 5, leSpeed);
+                pbf_press_dpad(context, DPAD_UP, 5, leSpeed);
+                i+=6;
+            }
+            else if(itemNo-i==5-itemOffset&&itemNo>=5)
+            {
+                pbf_press_dpad(context, DPAD_RIGHT, 5, leSpeed);
+                pbf_press_dpad(context, DPAD_UP, 5, leSpeed);
+                pbf_press_dpad(context, DPAD_UP, 5, leSpeed);
+                i+=5;
+            }
+            else
+            {
+                pbf_press_dpad(context, DPAD_DOWN, 5, 13);
+                i++;
+            }
+        }
+    }
+    else{
+        itemNo = totalItems-itemNo;
+        if(itemNo < 0){itemNo *= -1;}
+        pbf_press_dpad(context, DPAD_UP, 5, leSpeed);
+        i++;
+        while(i<itemNo){
+            context.wait_for_all_requests();
+            if(itemNo-i>=7&&itemNo>=7)
+            {
+                pbf_press_dpad(context, DPAD_LEFT, 5, leSpeed);
+                i+=7;
+            }
+            else if(itemNo-i==6&&itemNo>=6)
+            {
+                pbf_press_dpad(context, DPAD_LEFT, 5, leSpeed);
+                pbf_press_dpad(context, DPAD_DOWN, 5, leSpeed);
+                i+=6;
+            }
+            else if(itemNo-i==5&&itemNo>=5)
+            {
+                pbf_press_dpad(context, DPAD_LEFT, 5, leSpeed);
+                pbf_press_dpad(context, DPAD_DOWN, 5, leSpeed);
+                pbf_press_dpad(context, DPAD_DOWN, 5, leSpeed);
+                i+=5;
+            }
+            else
+            {
+                pbf_press_dpad(context, DPAD_UP, 5, 13);
+                i++;
+            }
+        }
+    }
+
+    pbf_wait(context, 10);
+    if(i>13){
+        pbf_wait(context, 35); // if there were too many moves the program will get a little dizzy, let's give it time.
+    }
+    context.wait_for_all_requests();
+    pbf_press_button(context, BUTTON_A, 5, 50);
+    context.wait_for_all_requests();
+}
+
+bool IsCursorRed(SingleSwitchProgramEnvironment& env, BotBaseContext& context)
+{
+    context.wait_for_all_requests();
+    for(int i = 0;i<3;i++){
+        if (!(DetectSelectionArrow(env.console.video().snapshot(),ImageFloatBox (.47, 0, 0.052, 0.012),0xffff4600,0xffe3d26c) != PokemonLA::ItemCompatibility::COMPATIBLE)){
+             return true;
+             break;
+        }
+        else{
+            pbf_press_button(context, BUTTON_ZR, 5, 150);
+        }
+        context.wait_for_all_requests();
+    }
+    return false;
+}
+
+void PokemonHomeSort(int pokedexNo,int letterNo,int movesDown,int movesUp,int upOrDown,int getShiny,int startBoxPage,int startBoxNo,BotBaseContext& context,SingleSwitchProgramEnvironment& env,int type1 = 0, int type2 = 0){
```

Comment:
```suggestion
void PokemonHomeSort(int pokedexNo,int letterNo,int movesDown,int movesUp,bool upOrDown,int getShiny,int startBoxPage,int startBoxNo,BotBaseContext& context,SingleSwitchProgramEnvironment& env,int type1 = 0, int type2 = 0){
```
this should be boolean instead
For type1 and type2, Could you use an enum too ? (there is a lot of parameters in this function)